sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -301,6 +301,8 @@ trait PrunedScan {
 @DeveloperApi
 trait PrunedFilteredScan {
   def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row]
+  def buildScan(
```

Comment:
this also breaks all existing implementations because we can't add a method to the interface.

Really - it is really hard to design a public interface that works for complicated cases (e.g. aggregation) and at the same time can be maintained in the years to come. If you are only interested in doing this for jdbc, I think we should consider some hacks to work around the public interface.